iwyu.cc

```diff
@@ -2712,7 +2646,83 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
   // void fn(const S<Class>& s) { // Forward declarations are sufficient here.
   //   (void)s.t; // Full 'Class' type is needed due to template instantiation.
   // }
-  void ReportTplSpecComponentTypes(const TemplateSpecializationType*) = delete;
+  void ReportTplSpecComponentTypes(const TemplateSpecializationType*,
+                                   const set<const Type*>& blocked_types) =
+      delete;
+
+  const std::set<const Type*>& GetBlockedTypes() const = delete;
```

Comment:
Let's use `set` without namespace qualification for new code. I saw there was some explicit qualification in existing code, but I can clean that up after.